cmd/servicelog/list.go

```diff
@@ -0,0 +1,125 @@
+package servicelog
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/openshift-online/ocm-cli/pkg/arguments"
+	"github.com/openshift-online/ocm-cli/pkg/dump"
+	sdk "github.com/openshift-online/ocm-sdk-go"
+	"github.com/openshift/osd-utils-cli/internal/servicelog"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+// listCmd represents the list command
+var listCmd = &cobra.Command{
+	Use:   "list",
+	Short: "gets all servicelog messages for a given cluster",
+	// validate only clusterid is provided
+	Args: cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		clusterId := args[0]
+
+		// Create an OCM client to talk to the cluster API
+		// the user has to be logged in (e.g. 'ocm login')
+		ocmClient := createConnection()
+		defer func() {
+			if err := ocmClient.Close(); err != nil {
+				log.Errorf("Cannot close the ocmClient (possible memory leak): %q", err)
+			}
+		}()
+
+		// Use the OCM client to create the POST request
+		request := createClusterRequest(ocmClient, clusterId)
+		response := sendRequest(request)
+		clusterExternalId, err := extractExternalIdFromResponse(response)
+		if err != nil {
+			return err
+		}
+
+		// send it as logservice and validate the response
+		request = createListRequest(ocmClient, clusterExternalId, serviceLogListFlags.allMessages)
+		response = sendRequest(request)
+
+		err = dump.Pretty(os.Stdout, response.Bytes())
+		if err != nil {
+			return err
+		}
+
+		return nil
+	},
+}
+
+var serviceLogListFlags = struct {
```

Comment:
since you have only one flag you could just have 
```
var allMessages bool
```
Once we get to support multiple flags, we can create a new struct

Reply:
fixed